opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -551,7 +551,7 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
             if (enableDissolvedGas()) {
                 // miscible oil
                 const LhsEval& Rs = Opm::BlackOil::template getRs_<ThisType, FluidState, LhsEval>(fluidState, regionIdx);
-                const LhsEval& bo = oilPvt_->inverseFormationVolumeFactor(regionIdx, T, p, Rs);
+                const LhsEval& bo = inverseFormationVolumeFactor(fluidState, phaseIdx, regionIdx);
```

Comment:
Isn't the new version identical to the old one? if no, the `Rs` variable above should be unused?!

Reply:
Do not think so, since since the member function do the bookkeeping  with of finding if the fluid is saturated or not. It probably had been better to take all of this directly from the state, since this will ensure consistency with the flow problem which is essential for physical behavoir.. (Else the mass in the energy equation is not consitent with the mass int the system and tempreratur can be gennerated.)

Reply:
How could I take it directly form the state. (I tried but broke some tests)